### PR TITLE
Add bool constructor to TextOf

### LINKED ISSUE #534: Add bool constructor to TextOf
### Expected Behavior
```TextOf``` can serialize bool values.

### Actual Behavior
```TextOf``` only accepts numbers, strings and classes related to those things.

### Mention any other details that might be useful
Whenever i'm turning a bunch of data into strings, i can use ```TextOf``` for everything exept bool values, instead i have to use ```someBoolValue.ToString(CultureInfo.InvariantCulture)```, which is annoying.

### The log given by the failure.

### PR DESCRIPTION
**Pull request by DFU398**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

See #534

### What is the new behavior?

closes #534

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs [hunk 1]
```diff
@@ -210,6 +210,31 @@ public void ReadsFloatWithNumberFormatIntoText()
...
+        [Theory]
+        [InlineData(true, "True")]
+        [InlineData(false, "False")]
+        public void ReadsBoolIntoTextWithCultureInfo(bool input, string expected)
+        {
+            Assert.True(
+                new TextOf(
+                    input,
+                    new CultureInfo("en-US")
+                ).AsString() == expected,
```
Comment (pz1337) [inline tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs:233]: 💅🏻 Is there any culture that would change the outcome?

**Review by pz1337 — APPROVED** [review]

File: tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs (hunk 1 above)
Comment (koeeenig) [reply tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs:233]: I am not aware of any but didn't investigate too much.

**Review by koeeenig — APPROVED** [review]

File: tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs (hunk 1 above)
Comment (DFU398) [reply tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs:233]: I didn't investigate that either, but i think there is probably a reason why the bool.ToString method accepts a CultureInfo as argument.

Comment (pz1337) [reply tests/Yaapii.Atoms.Tests/Text/TextOfTest.cs:233]: Ok, thanks :)

**Pull request merged**